workers_manager.py

```diff
@@ -100,6 +110,33 @@ def _pip_install_helper(package_names):
     for package in package_names:
       pip_main(['install', '-q', package])
 
+  def _poll_wrapper(self, worker_obj, client, userdata, c):
+    if c.payload.decode('utf-8') != '':
+      return
+    poll_device_supported = 'poll_device' in worker_obj.status_update.__code__.co_varnames
+    topic_prefixes = []
+    topic_prefixes.append(userdata.get('global_topic_prefix', ''))
+    topic_prefixes.append(getattr(worker_obj, 'topic_prefix', ''))
+    topic_prefixes.append('poll')
+    device_name = str(c.topic)
+    for suffix in topic_prefixes:
+      if suffix:
+        device_name = str(device_name).replace(suffix, '').replace('/', '')
+    device_name = device_name if device_name else None
+    if not poll_device_supported and device_name:
+      _LOGGER.debug("This worker does not support polling specific devices."
+                    " Ignoring: %s", c.topic)
+      return
+    elif not poll_device_supported:
+      poll_args = []
+    elif device_name not in getattr(worker_obj, 'devices', []):
```

Comment:
Please add some getter for devices and use it, instead of accessing variable directly.